pallets/services-payment/src/lib.rs

```diff
@@ -344,14 +292,35 @@ impl<T: Config> AuthorNotingHook<T::AccountId> for Pallet<T> {
     ) -> Weight {
         let total_weight = T::DbWeight::get().reads_writes(1, 1);
 
-        if let Err(e) = Pallet::<T>::burn_credit_for_para(&para_id) {
-            log::warn!(
-                "Failed to burn credits for container chain {}: {:?}",
-                u32::from(para_id),
-                e
-            );
+        if Pallet::<T>::burn_credit_for_para(&para_id).is_err() {
+            let (amount_to_charge, weight) = T::ProvideBlockProductionCost::block_cost(&para_id);
+            match T::Currency::withdraw(
+                &Self::parachain_tank(para_id),
+                amount_to_charge,
+                WithdrawReasons::FEE,
+                ExistenceRequirement::AllowDeath,
+            ) {
+                Err(e) => log::warn!(
+                    "Failed to withdraw credits for container chain {}: {:?}",
+                    u32::from(para_id),
+                    e
+                ),
+                Ok(imbalance) => {
+                    T::OnChargeForBlock::on_unbalanced(imbalance);
+                }
+            }
+            total_weight.saturating_add(weight);
```

Comment:
This branch is also missing the weight of `T::Currency::withdraw` and `T::OnChargeForBlock::on_unbalanced`. Not sure how I would benchmark this.

Also add a comment somewhere that says something like: "First try to burn a free credit, if there are no credits then use the parachain tank account"

Reply:
yes will do, and I am literally going to benchmark on_author_noted entirely